src/llms/openai.ts

```diff
@@ -0,0 +1,345 @@
+/**
+ * OpenAI API client.
+ *
+ * This module contains functions used to make requests to the OpenAI API via
+ * the Grafana LLM app plugin. That plugin must be installed, enabled and configured
+ * in order for these functions to work.
+ *
+ * The {@link enabled} function can be used to check if the plugin is enabled and configured.
+ */
+
+import { isLiveChannelMessageEvent, LiveChannelAddress, LiveChannelMessageEvent, LiveChannelScope } from "@grafana/data";
+import { getBackendSrv, getGrafanaLiveSrv, logDebug } from "@grafana/runtime";
+
+import { pipe, Observable, UnaryFunction } from "rxjs";
+import { filter, map, scan, takeWhile } from "rxjs/operators";
+
+const LLM_PLUGIN_ID = 'grafana-llm-app';
+const LLM_PLUGIN_ROUTE = `/api/plugins/${LLM_PLUGIN_ID}`;
+const OPENAI_CHAT_COMPLETIONS_PATH = 'openai/v1/chat/completions';
```

Comment:
```suggestion
const OPENAI_CHAT_COMPLETIONS_PATH = '/v1/chat/completions';
```
Should be just /v1 no?

Reply:
This needs to have the `openai` on it still to identify it as an OpenAI call. I've just fixed the app backend to correctly strip off `openai` rather than `/openai`, so it should work now!